src/main/java/uk/gov/ons/validation/service/ProcessWranglerQuestionData.java

```diff
@@ -0,0 +1,148 @@
+                                invokeConfig.getFinalDerivedQuestValue()));
+                        //Call External Lambda which performs Validation
+                        callValidationLambda(invokeConfig);
+                        break;
+                    }
+                }
+            }
+        }
+        // BPM (Business Process Management) response goes here. This method puts 
+        sendBpmResponse("B123-P456-M789", "QvsDQ", "arn:aws:sns:eu-west-2:014669633018:Take-On-Validation-SNS");
+    }
+
+    /**
+     * Call Validation Lambda i.e. VET
+     *
+     * @param config InvokeConfig
+     * @throws JsonProcessingException
+     */
+    private void callValidationLambda(InvokeConfig config) throws JsonProcessingException {
+        try {
+            //Call External Lambda which performs Validation
+            log.info("Before Calling Validation Lambda");
+            WranglerResponseData dataElement = WranglerResponseData.builder()
+                    .primaryValue(config.getFinalQuestCodeValue())
+                    .comparisonValue(config.getFinalDerivedQuestValue())
+                    .build();
+            String requestJson = new ObjectMapper().writeValueAsString(dataElement);
+            String wranglerName = PropertiesUtil.getProperty(Constants.WRANGLER_NAME);
+            log.info(format(SEND_MESSAGE, wranglerName, requestJson));
+            InvokeRequest invokeRequest = newInvokeRequest();
+            invokeRequest.withFunctionName(wranglerName).withPayload(requestJson);
+            InvokeResult result = buildAWSLambdaClient().invoke(invokeRequest);
+            log.info(format("Status after calling Validation Lambda %s", result.getStatusCode()));
+        } catch (JsonProcessingException e) {
+            log.error("An exception occurred while attempting to prepare " +
+                    "and send the request to Validation lambda.", e);
+            throw e;
+        }
+    }
+
+    /**
+     * Build AWS Lambda Client
+     *
+     * @return AWSLambdaAsync
+     */
+    private AWSLambdaAsync buildAWSLambdaClient() {
+        return AWSLambdaAsyncClient.asyncBuilder().withRegion(Regions.EU_WEST_2).build();
+    }
+
+    /**
+     * Create new InvokeRequest for Lambda
+     *
+     * @return InvokeRequest
+     */
+    private InvokeRequest newInvokeRequest() {
+        return new InvokeRequest();
+    }
+
+    private String constructBpmResponse(String bpmInstance, String validationName) throws JsonProcessingException{
+        MessageResponse message = MessageResponse.builder().bpmInstance(bpmInstance).validationName(validationName).build();
+        ObjectMapper objectMapper = new ObjectMapper();
+        return objectMapper.writeValueAsString(message);
+    }
+
+    private void sendBpmResponse(String bpmInstance, String validationName, String topicArn) throws JsonProcessingException{
+        String messageToSend;
+        try{
```

Comment:
Could be done in 1 try/catch, or two methods (one to construct and one to publish if you want to keep the two try/catch statements.

Reply:
Unless there's a compelling reason to change this, I'm going to keep it as it is right now.